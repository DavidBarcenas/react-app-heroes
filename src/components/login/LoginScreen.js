import React, { useContext } from 'react'
import './login.css'
import { AuthContext } from '../../auth/AuthContext'
import { types } from '../../types/types'

export const LoginScreen = ({ history }) => {

  const {dispatch} = useContext(AuthContext)
  const lastPath = localStorage.getItem('lastPath') || '/';

  const handleClick = () => {
    // history.push('/')
    dispatch({
      type: types.login,
      payload: {
        name: 'Daveepro'
      }
    })

    history.push(lastPath)
  }

  return (
    <div className="login__wrapper">
      <div className="login__box card">

      <img src="./assets/lgo.jpg" alt="MarvelDc" className="login__logo" />
      <button className="login__btn btn red waves-effect waves-light" onClick={ handleClick }>Login</button>
      </div>
    </div>
  )
}
