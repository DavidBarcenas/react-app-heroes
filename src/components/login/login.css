.login__wrapper {
  width: 100vw;
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
}

.login__box {
  width: 250px;
  padding: 1em;
}

.login__logo {
  width: 100%;
  object-fit: cover;
}

.login__btn {
  display: block;
  width: 100%;
}