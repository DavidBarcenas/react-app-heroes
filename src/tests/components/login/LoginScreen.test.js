import '@testing-library/jest-dom';
import React from 'react'
import { mount } from 'enzyme';
import { LoginScreen } from '../../../components/login/LoginScreen';
import { AuthContext } from '../../../auth/AuthContext';
import { types } from '../../../types/types';


describe('<LoginScreen /> testing', () => {

  const contextValue = {
    dispatch: jest.fn()
  }
  
  const history = {
    replace: jest.fn()
  }

  afterEach(() =>{
    jest.clearAllMocks()
  })
  
  test('should render correctly', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <LoginScreen />
      </AuthContext.Provider>
    )
    
    expect(wrapper).toMatchSnapshot()
  })
  
  test('should run dispatch and navigation', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <LoginScreen history={history} />
      </AuthContext.Provider>
    )
    const handleClick = wrapper.find('button').prop('onClick');
    handleClick()

    expect(contextValue.dispatch).toHaveBeenCalledWith({
      type: types.login,
      payload: {
        name: 'Daveepro'
      }
    })
    expect(history.replace).toHaveBeenCalledWith('/')
    
    localStorage.setItem('lastPath', '/dc');
    handleClick()



    expect(history.replace).toHaveBeenCalledWith('/dc')
  })
  
})
