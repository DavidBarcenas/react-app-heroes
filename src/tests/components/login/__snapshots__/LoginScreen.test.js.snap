// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginScreen /> testing should render correctly 1`] = `
<div
  className="login__wrapper"
>
  <div
    className="login__box card"
  >
    <img
      alt="MarvelDc"
      className="login__logo"
      src="./assets/lgo.jpg"
    />
    <button
      className="login__btn btn red waves-effect waves-light"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
